### PR TITLE
pubsys: unwrap AWS SDK errors in SSM calls

### PR DESCRIPTION
**Pull request by rpkelly**
**Description of changes:**
This surfaces the AWS API service error codes in error log message when using pubys to interact with SSM.

**Testing done:**

- [x] ran pubsys to successful create and update SSM parameters in my account
- [x] ran pubsys to update SSM in an opt in region my account does not have access to in order to trigger an error

**Terms of contribution:**

By submitting this pull request, I agree that this contribution is dual-licensed under the terms of both the Apache License, version 2.0, and the MIT license.

### CODE REVIEW

**Review by jpculp — APPROVED** [review]

**Review by etungsten — APPROVED** [review]

File: tools/pubsys/src/aws/ssm/ssm.rs [hunk 1]
```diff
@@ -239,7 +239,12 @@ pub(crate) async fn set_parameters(
                 // RusotoError::Unknown(BufferedHttpResponse {status: 400, body: "{\"__type\":\"ThrottlingException\",\"message\":\"Rate exceeded\"}", headers: ...})
                 // Even if we were to do a structural match, we would still have to string match
                 // the body of the error.  Simpler to match the string form.
-                if e.to_string().contains("ThrottlingException") {
```
Comment (webern) [inline tools/pubsys/src/aws/ssm/ssm.rs:242]: I can't comment on lines 239-241 but the comment mentions Rusoto and is probably outdated.

File: tools/pubsys/src/aws/ssm/ssm.rs [hunk 2]
```diff
@@ -239,7 +239,12 @@ pub(crate) async fn set_parameters(
...
+                let error_type = e
+                    .into_service_error()
+                    .code()
```
Comment (webern) [inline tools/pubsys/src/aws/ssm/ssm.rs:241]: I tried for 30 minutes to find the enum that is providing this string message without success. I guess the stringly typing is the best we can do given how sprawling the library is.

**Review by webern — APPROVED** [review]

File: tools/pubsys/src/aws/ssm/ssm.rs (hunk 1 above)
Comment (rpkelly) [reply tools/pubsys/src/aws/ssm/ssm.rs:242]: Fixed the comment

File: tools/pubsys/src/aws/ssm/ssm.rs (hunk 2 above)
Comment (rpkelly) [reply tools/pubsys/src/aws/ssm/ssm.rs:241]: From what I can tell, Throttling Exceptions are not currently surfaced: https://github.com/aws/aws-sdk/issues/362

**Review by jpculp — APPROVED** [review]

**Pull request merged**